llvm/lib/Target/DirectX/DXILShaderFlags.cpp

```diff
@@ -36,8 +39,35 @@ static void updateFlags(ComputedShaderFlags &Flags, const Instruction &I) {
   }
 }
 
-ComputedShaderFlags ComputedShaderFlags::computeFlags(Module &M) {
+static void updateResourceFlags(ComputedShaderFlags &Flags, Module &M,
+                                ModuleAnalysisManager *AM) {
+  if (!AM)
+    return;
+
+  const DXILResourceMap &DRM = AM->getResult<DXILResourceAnalysis>(M);
+  if (DRM.empty())
+    return;
+
+  const dxil::ModuleMetadataInfo &MMDI = AM->getResult<DXILMetadataAnalysis>(M);
+  VersionTuple SM = MMDI.ShaderModelVersion;
+  Triple::EnvironmentType SP = MMDI.ShaderProfile;
+
+  // RWBuffer
+  for (const ResourceInfo &RI : DRM.uavs()) {
+    if (RI.getResourceKind() == ResourceKind::TypedBuffer) {
+      Flags.EnableRawAndStructuredBuffers = true;
+      Flags.ComputeShadersPlusRawAndStructuredBuffers =
+          (SP == Triple::EnvironmentType::Compute && SM.getMajor() == 4);
```

Comment:
```SM.getMajor() == 4```

This means SM 4.x?  In which case this flag will never be set.  I'm surprised it isn't `>= 4`.

Reply:
Yeah.. I was too!  I looked at the DXC source and it does this type of check.. Since I recently learned that there is no way we would even support SM4, we should just not set this flag anyway and remove it.

Reply:
Probably should get some more opinions on this.  If we were to ever support SM 4.0 via DXBC in the future then having this in here would likely be helpful to that effort.

Reply:
This issue has been resolved in chat.  The result is to not set this flag as it will not be used at this time.